### PR TITLE
[Fix] Avoid calling `model['args']` in yolov8 validation.

### PR DESCRIPTION
**Pull request by dbogunowicz**
Fix for: https://app.asana.com/0/1203126676641557/1204141253385896
Fix for: https://app.asana.com/0/1203126676641557/1204130698270567

## Fix description
The validation script is not compatible with the training script for yolov8 in one regard, that the loaded model does not have `args` attribute. However, the same data is present in the model, albeit in a different place. 

## Testing
1. With custom sparsified model:
Trained the sparse and quantized model that is saved under: `runs/detect/train102/weights/last.pt`

```bash
sparseml.ultralytics.val --model yolov8n
```

```bash
Applying structure from sparseml checkpoint at epoch -1
Loaded previous weights from checkpoint
Ultralytics YOLOv8.0.30 🚀 Python-3.8.10 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4000, 16117MiB)
val: Scanning /home/ubuntu/damian/deepsparse/datasets/coco128/labels/train2017.cache... 126 images, 2 backgrounds, 0 corrupt: 100%|██████████| 128/128 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:01<00:00,  6.05it/s]
                   all        128        929          0          0          0          0
Speed: 0.1ms pre-process, 4.0ms inference, 0.0ms loss, 0.0ms post-process per image
```

(rubbish accuracy makes sense, this model does not have any sensible weights)
2. With vanilla model `yolov8n`

```bash
sparseml.ultralytics.val --model yolov8n
```

Output:

```
Ultralytics YOLOv8.0.30 🚀 Python-3.8.10 torch-1.13.1+cu117 CUDA:0 (NVIDIA RTX A4000, 16117MiB)
val: Scanning /home/ubuntu/damian/deepsparse/datasets/coco128/labels/train2017.cache... 126 images, 2 backgrounds, 0 corrupt: 100%|██████████| 128/128 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 8/8 [00:02<00:00,  3.46it/s]
                   all        128        929      0.641      0.537      0.605      0.446
Speed: 0.1ms pre-process, 1.9ms inference, 0.0ms loss, 0.8ms post-process per image
```

### CODE REVIEW

**Review by corey-nm — APPROVED** [review]
lgtm!

**Review by abhinavnmagic — CHANGES_REQUESTED** [review]
1. This PR doesn't fix pytorch checkpoints obtained using one-shot. I will post the error in the asana ticket.
2. The obtained metrics do not match the wandb numbers. Let's talk about this.
